aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,956 @@
+
+    /// Entry function that can be called by anyone
+    public entry fun resolve<CoinType>(proposal_id: u64, nft_dao: address) acquires Proposals, DAO, ProposalVotingStatistics {
+        // validate if proposal is ready to resolve
+        let dao = borrow_global_mut<DAO>(nft_dao);
+
+        // assert the proposal voting ended
+        let proposals = borrow_global_mut<Proposals>(nft_dao);
+        let proposal = table::borrow_mut(&mut proposals.proposals, proposal_id);
+        let now = timestamp::now_seconds();
+        assert!(now >= proposal.start_time_sec + dao.voting_duration, error::invalid_argument(EPROPOSAL_ENDED));
+
+        // assert the proposal is unresolved yet
+        assert!(proposal.resolution == PROPOSAL_PENDING, error::invalid_argument(EPROPOSAL_RESOLVED));
+
+        let proposal_stat = &mut borrow_global_mut<ProposalVotingStatistics>(nft_dao).proposals;
+        let voting_stat = table::borrow_mut(proposal_stat, proposal_id);
+        proposal.final_yes_votes = voting_stat.total_yes;
+        proposal.final_no_votes = voting_stat.total_no;
+        // validate resolve threshold and result
+        let voted = voting_stat.total_no + voting_stat.total_yes;
+        if (voted < dao.resolve_threshold) {
+            // not sufficient token voted
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        } else if(voting_stat.total_yes > voting_stat.total_no) {
+            let function_name = proposal.function_name;
+            if (function_name == string::utf8(b"transfer_fund")) {
+                let res_signer =  create_signer_with_capability(&dao.dao_signer_capability);
+                let dst_addr = property_map::read_address(&proposal.function_args, &string::utf8(b"dst"));
+                let amount = property_map::read_u64(&proposal.function_args, &string::utf8(b"amount"));
+                transfer_fund<CoinType>(&res_signer, dst_addr, amount);
+            } else {
+               assert!(function_name == string::utf8(b"no_op"), error::invalid_argument(ENOT_SUPPROTED_FUNCTION));
+            };
+            proposal.resolution = PROPOSAL_RESOLVED_PASSED;
+        } else {
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        };
+
+        nft_dao_events::emit_resolve_event(
+            proposal_id,
+            nft_dao,
+            proposal.resolution,
+        )
+    }
+
+    /// Offer DAO to an address
+    public entry fun offer_dao(admin: &signer, dao: address, new_admin: address) acquires DAO, OwnedDAO, DAOOffer {
+        let admin_addr = signer::address_of(admin);
+        let dao_config = borrow_global<DAO>(dao);
+        assert!(admin_addr == dao_config.admin, error::permission_denied(EINVALID_ADMIN_ACCOUNT));
+        let owned_dao = &borrow_global<OwnedDAO>(admin_addr).daos;
+        let (found, _) = vector::index_of(owned_dao, &dao);
+        assert!(found, error::not_found(ENOT_IN_ADMIN_OWNED_DAO_ADDRESSES));
+
+        if(!exists<DAOOffer>(admin_addr)){
```

Comment:
I think it might be more appropriate to store the pending transfer in the DAO resource account instead of admin's. This would make it more visible to the dao community who's looking at the DAO account instead of the admin's. We shouldn't store anything on the admin's account, except for ownedDAOs, which is only for reference convenience.

Reply:
Speaking of OwnedDaos, is this a feature requested for Aries? In my experience, most admin accounts are multisigs and are only admin in exactly 1 DAO. I've never seen an account who's admin of multiple. In that case, OwnedDAOs is not really necessary at all and can be removed for simplicity.

Reply:
There are two places storing the admin. One is in the OwnedDAO and the other is in DAO struct itself.  The change is visible to dao community since the DAO struct is stored on the resource account. I will add an event emitting from the DAO for this change.  On the other hand, OwnedDAO is just for the admin of DAO to keep track of the DAOs she owns. Without it, admin needs to rely on an indexer to get the DAO he owns, which is not ideal

Having a list of dao addresses in OwnedDAO, is to make it less restrictive. For example, is it necessary to enforce each user can only create one DAO on the Aries platform?

Reply:
I don’t think the community really first looks at the admin address. They would start from the dao address and find out who the admin is. The only benefit of having owned dao imo is if the UI needs it so that when the admin logs into Aries for example, they see the dao loaded instantly. But that can also be done via events. Hence my question about whether Aries explicitly asks for this

Reply:
How would the service know which event stream to look when the admin logs in?

Reply:
deleted this owneddao based on offline discussion